crates/cairo-coverage/src/output/lcov.rs

```diff
@@ -27,24 +29,27 @@ struct LcovDetails {
 
 impl From<FilesCoverageData> for LcovFormat {
     fn from(files_coverage_data: FilesCoverageData) -> Self {
-        let mut list: Vec<_> = files_coverage_data
-            .iter()
-            .map(|(file_location, file_coverage_data)| {
-                (file_location.to_owned(), file_coverage_data.into())
-            })
-            .collect();
-        list.sort_by_key(|(file_location, _)| file_location.to_owned());
-        Self(list)
+        Self(
+            files_coverage_data
+                .iter()
+                .map(|(source_file_full_path, file_coverage_data)| {
+                    (source_file_full_path.to_owned(), file_coverage_data.into())
+                })
+                .sorted_by_key(|(source_file_full_path, _)| source_file_full_path.clone())
```

Comment:
what is an impact of multiple cloning in here? I'm particularly worried about larger projects. Maybe source_file_full_path could be referrenced during sorting?

Reply:
I'm uncertain about the performance. It essentially involves cloning a string. We can do something like this to avoid cloning:

```rust
.sorted_by(
    |(source_file_full_path, _), (other_source_file_full_path, _)| {
        source_file_full_path.cmp(other_source_file_full_path)
    },
)
```

Although this is less clean.

Reply:
perfect, thanks :+1: